keras_cv/layers/preprocessing/posterization_test.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.posterization import Posterization
+
+
+class PosterizationTest(tf.test.TestCase):
+    rng = tf.random.Generator.from_seed(1234)
+
+    def test_raises_error_on_invalid_bits_parameter(self):
+        invalid_values = [-1, 0, 9, 24]
+
+        for value in invalid_values:
+            with self.assertRaises(AssertionError):
+                Posterization(bits=value)
+
+    def test_output_shape_unchanged(self):
```

Comment:
@LukeWood We are writing this quite the same in every class. Is there any refactor solution?

Reply:
we are working on exporting a parameterized `layer_test` method that extends `tf.TestCase` from keras core.  This will cover most of the generic behaviors.